lib/src/metta/runner/mod.rs

```diff
@@ -241,11 +241,43 @@ impl Metta {
         self.0.settings.borrow().get(key.into()).map(|a| a.to_string())
     }
 
+    /// Runs a MeTTa program expressed as a string of characters
+    pub fn run_program_str(&self, program: &str) -> Result<Vec<Vec<Atom>>, String> {
+        let parser = SExprParser::new(program);
+        self.run(parser)
+    }
+
     pub fn run<'p, 'a: 'p>(&'a self, parser: SExprParser<'p>) -> Result<Vec<Vec<Atom>>, String> {
         let state = RunnerState::new_with_parser(self, parser);
         state.run_to_completion()
     }
 
+    /// Parses a string into a single Atom
+    pub fn parse_one_atom(&self, text: &str) -> Result<Atom, String> {
```

Comment:
I would replace usages in `sorted_list.rs` by macros and move the code of the method to the `test_utils.rs`.

Reply:
I explained my reasoning behind the `parse_all_atoms` public API above.

The main reason I added a second method here.  (`parse_one_atom`) was that I didn't want the overhead of creating a Vec<> for each trip through a loop that consumes atoms one at a time.  This seemed like it might be a common pattern for a caller.

However, I don't feel as strongly about this, and I wouldn't object to making it private or rolling it into its caller(s)

Reply:
API removed in https://github.com/trueagi-io/hyperon-experimental/commit/28e464e3e807741a2309278411dcf49fd8fb84d9  See comments above.